src/features/datasets/components/DatasetForm.tsx

```diff
@@ -0,0 +1,152 @@
+import { Button } from "@/src/components/ui/button";
+import * as z from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/src/components/ui/form";
+import { api } from "@/src/utils/api";
+import { useState } from "react";
+import { usePostHog } from "posthog-js/react";
+import { Input } from "@/src/components/ui/input";
+
+interface BaseDatasetFormProps {
+  mode: "create" | "rename";
+  projectId: string;
+  onFormSuccess?: () => void;
+}
+
+interface CreateDatasetFormProps extends BaseDatasetFormProps {
+  mode: "create";
+}
+
+interface RenameDatasetFormProps extends BaseDatasetFormProps {
+  mode: "rename";
+  datasetId: string; // Make datasetId non-optional for 'rename' mode
+  datasetName: string;
+}
+
+type DatasetFormProps = CreateDatasetFormProps | RenameDatasetFormProps;
+
+const formSchema = z.object({
+  name: z
+    .string()
+    .min(1, { message: "Input is required" })
+    .refine((name) => name.trim() === name, {
+      message: "Input should not have leading or trailing whitespace",
+    })
+    .refine((name) => name.trim().length > 0, {
+      message: "Input should not be only whitespace",
+    }),
+});
```

Comment:
Thanks for considering the edge cases here. What do you think of just trimming the input for a user? I doubt that someone wants to intentionally add a whitespace prefix or suffix.